### ISSUE TITLE
CVC4 hangs when I trying to get proof

### ISSUE THREAD

**Issue by j123123** [CVC4/CVC4#2843]
CVC4 hangs when I trying to get proof:
````SMT
(set-logic ALL)
(set-option :produce-models true)
(set-option :produce-proofs true)

(declare-const a Int)

(assert (and 
          (=
            a
            (* a a)
          )
          (not (= a 0))
          (not (= a 1))
        )
)

(check-sat)
(get-model)
(get-proof)


(exit)
````

got infinity output, which looks like (R _ _ .lemc12s59286 (R _ _ .lemc12s59287 (R _ _ .lemc12s59288 (R _ _ .lemc12s59289 (R _ _ .lemc12s59290 (R _ _ .lemc12s59291 (R _ _ .lemc12s59292 (R _ _ .lemc12s59293 (R _ _ .lemc12s59294 (R _ _ .lemc12s59295 (R _ _ .lemc12s59296 (R _ _ .lemc12s59297 (R _ _ .lemc12s59298 (R _ _ .lemc12s59299 (R _ _ .lemc12s59300 (R _ _ .lemc12s59301 (R _ _ .lemc12s59302 (R _ _ .lemc12s59303 (R _ _ .lemc12s59304 (R _ _ .lemc12s59305 (R _ _ .lemc12s59306 (R _ _ .lemc12s59307 (R _ _ .lemc12s59308 (R _ _ .lemc12s59309 (R _ _ .lemc12s59310 (R _ _ .lemc12s59311 (R _ _ .lemc12s59312 ...

CVC4 version 1.6 compiled from source taken from http://cvc4.cs.stanford.edu/downloads/builds/src/cvc4-1.6.tar.gz

**Comment by 4tXJ7f** [comment]
Thank you for the report, I'll take a look.

**Comment by 4tXJ7f** [comment]
I just realized that you are using non-linear integer arithmetic. Sadly, CVC4 does not support proofs for QF_NIA yet. CVC4 1.6 does not support any arithmetic proofs at all. Current master supports some arithmetic proofs based on [Farkas' Lemma](https://en.wikipedia.org/wiki/Farkas%27_lemma). Your input crashes current master, so I'll fix that but proofs for QF_NIA is a feature that we don't currently have a time frame for, unfortunately.

**Comment by yoni206** [comment]
@4tXJ7f, this bug is resolved by #3327, isn't it? In particular, I am now getting an appropriate error message about unsupported logics for proofs when running the above benchmark. May I close this?

**Comment by 4tXJ7f** [comment]
Yup, that should be good!

**Comment by yoni206** [comment]
resolved by #3327 .
